docs/providers/aws/events/cloudfront.md

```diff
@@ -0,0 +1,161 @@
+<!--
+title: Serverless Framework - AWS Lambda Events - CloudFront
+menuText: CloudFront
+menuOrder: 13
+description:  Setting up CloudFront with AWS Lambda@Edge via the Serverless Framework
+layout: Doc
+-->
+
+<!-- DOCS-SITE-LINK:START automatically generated  -->
+
+### [Read this on the main serverless docs site](https://www.serverless.com/framework/docs/providers/aws/events/cloudfront)
+
+<!-- DOCS-SITE-LINK:END -->
+
+# CloudFront
+
+[Amazon CloudFront](https://aws.amazon.com/cloudfront) is a content delivery network (CDN) service that allows Lambda functions to be executed at edge locations.
+
+Distribution configuration contains origins and behaviors which are used to define how to cache and deliver content from other services.
+
+Origin is the endpoint definition of the service that is delivered, e.g. S3 bucket or a website.
+
+Behavior defines how the Amazon CloudFront acts when the request hits the service. That is where Lambda@Edge functions are also defined.
+
+Lambda@Edge has four options when the Lambda function is triggered
+
+- `viewer-request`, when the CloudFront first receives the request from the client
+- `origin-request`, before the request to the origin service
+- `origin-response`, when CloudFront receives the response from the origin service
+- `viewer-response`, before the response returned to the client
+
+**IMPORTANT** Due to current [Lambda@Edge limitations](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-edge-delete-replicas.html) it's necessary to set the `DeletionPolicy` to `Retain` for AWS Lambda functions which use the `cloudFront` event. The Serverless Framework will do this automatically for you. However bear in mind that **you have to delete those AWS Lambda functions manually** once you've removed the service via `serverless remove`.
```

Comment:
Yes, good question :+1:

I tested it and AWS won't remove the Lambda once the CloudFront distribution is removed. I guess that's because the AWS Lambda function is still its own entity compared to the replicas which are bound to the Lambda in a 1:n relationship.

Replicas will be automatically removed and once that happened, you can remove the Lambda (you can't immediately remove the Lambda once the stack removal was successful).

Reply:
Ok, thanks for clarification.

Indeed currently we don't have means to address  that cleanly.

Still if at some point we will have native menu bar app, then deletion of that lambda might be delegated to it (another nice use case for an app)

Reply:
Sounds good! :+1: